cdci_data_analysis/analysis/drupal_helper.py

```diff
@@ -0,0 +1,367 @@
+
+
+def post_observation(product_gallery_url, jwt_token, t1=None, t2=None):
+    # post new observation with or without a specific time range
+    body_gallery_observation_node = copy.deepcopy(body_article_product_gallery.body_article)
+    # set the type of content to post
+    body_gallery_observation_node["_links"]["type"]["href"] = body_gallery_observation_node["_links"]["type"][
+                                                                  "href"] + 'observation'
+    if t1 is not None and t2 is not None:
+        # format the time fields, from the format request
+        t1_formatted = parser.parse(t1).strftime('%Y-%m-%dT%H:%M:%S')
+        t2_formatted = parser.parse(t2).strftime('%Y-%m-%dT%H:%M:%S')
+        # set the daterange
+        body_gallery_observation_node["field_timerange"] = [{
+            "value": t1_formatted,
+            "end_value": t2_formatted
+        }]
+
+        body_gallery_observation_node["title"]["value"] = "_".join(["observation", t1_formatted, t2_formatted])
+    else:
+        # assign a randomly generate id in case to time range is provided
+        body_gallery_observation_node["title"]["value"] = "_".join(["observation", str(uuid.uuid4())])
+
+    headers = {
+        'Content-type': 'application/hal+json',
+        'Authorization': 'Bearer ' + jwt_token
+    }
+    # post the article
+    log_res = requests.post(f"{product_gallery_url}/node?_format=hal_json",
+                            data=json.dumps(body_gallery_observation_node),
+                            headers=headers
+                            )
+    output_post = log_res.json()
+    if log_res.status_code < 200 or log_res.status_code >= 300:
+        raise RequestNotUnderstood(output_post['message'],
+                                   status_code=log_res.status_code,
+                                   payload={'error_message': 'error while posting a new observation'})
+
+    # extract the id of the observation
+    observation_drupal_id = output_post['nid'][0]['value']
+
+    return observation_drupal_id
+
+
+def get_observation_drupal_id(product_gallery_url, jwt_token, t1=None, t2=None, observation_id=None):
+    observation_drupal_id = None
+    if observation_id is not None:
+        # get from the drupal the relative id
+        headers = {
+            'Content-type': 'application/hal+json',
+            'Authorization': 'Bearer ' + jwt_token
+        }
+        # post the article
+        log_res = requests.get(f"{product_gallery_url}/observations/{observation_id}?_format=hal_json",
+                               headers=headers
+                               )
+        output_get = log_res.json()
+        if log_res.status_code < 200 or log_res.status_code >= 300:
+            raise RequestNotUnderstood(output_get['message'],
+                                       status_code=log_res.status_code,
+                                       payload={'error_message': 'error while retrieving the user id'})
```

Comment:
message not right.

by the way, I suspect the reason why the message and the comments are occasionally not right is because it is copied and modified :)
So if the repetition was avoided, by putting the authorization and the error detection in a function, it would prevent this cases. But really as you wish!